src/elements/content-sidebar/activity-feed/task-form/TaskForm.js

```diff
@@ -126,31 +131,44 @@ class TaskForm extends React.Component<Props, State> {
         this.validateForm();
     };
 
-    handleCreateSuccess = () => {
-        const { onCreateSuccess } = this.props;
-        if (onCreateSuccess) {
-            onCreateSuccess();
+    handleSubmitSuccess = () => {
+        const { onSubmitSuccess } = this.props;
+        if (onSubmitSuccess) {
+            onSubmitSuccess();
         }
 
         this.clearForm();
         this.setState({ isLoading: false });
     };
 
-    handleCreateError = (e: ElementsXhrError) => {
-        const { onCreateError } = this.props;
-        onCreateError(e);
+    handleSubmitError = (e: ElementsXhrError) => {
+        const { onSubmitError } = this.props;
+        onSubmitError(e);
         this.setState({ isLoading: false });
     };
 
     handleValidSubmit = (): void => {
-        const { createTask, taskType } = this.props;
+        const { id, createTask, editTask, editMode, taskType } = this.props;
         const { message, approvers, dueDate, isValid } = this.state;
         const dueDateString = dueDate && dueDate.toISOString();
 
         if (!isValid) return;
 
         this.setState({ isLoading: true });
-        createTask(message, approvers, taskType, dueDateString, this.handleCreateSuccess, this.handleCreateError);
+
+        if (editMode === TASK_EDIT_MODE_EDIT && editTask) {
+            editTask(
```

Comment:
I think `editTask` and `createTask` should have the same signature. Do we want to update `createTask` to use an object? or should `editTask` be flattened?

Reply:
Object might be easier to deal with a flowType.